core/src/env/srml/calls.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::env::EnvTypes;
+use parity_codec::{
+    Decode,
+    Encode,
+    Input,
+    Output,
+};
+use core::convert::TryInto;
+
+#[cfg_attr(feature = "test-env", derive(Debug, Clone, PartialEq, Eq))]
```

Comment:
Is `test-env` correct here? As already said I think this could even life under its very own crate feature to make the `node-runtime` and `srml_indices` dev-dependencies optional upon testing. Maybe something like `test-dispatch-call` crate feature - use your creativity. Also please tell me what you think about this.

Reply:
Agree, this has now moved to https://github.com/paritytech/ink-types-node-runtime/pull/3. Over there the codec tests are the only ones so we need to build the substrate deps everytime anyway.